CLI: Detect vite project, use vite builder automatically

IanVS opened the pull request:
Issue:

## What I did

This is intended to aid discoverability of the vite builder by automatically initializing a project with the vite builder if a vite config file is detected.  It uses a similar approach of looking for a vite config file as [vitest](https://github.com/vitest-dev/vitest/blob/382aa351421d378c4b16301b7d94184e061e9a72/packages/vitest/src/node/create.ts#L16).

I could see also detecting whether to use the webpack5 builder, but I didn't do that here, it will fall back to webpack4, as before.  

If a `--builder` option is explicitly provided, it will be used and autodetection won't run.

This also cleans up a few other things, which I can remove if desired:

1) No longer logs `Not using community vite builder, skipping` after `init`, since the distinction between the community vite builder and official vite builder is subtle and might be confusing to users setting up a new storybook project.
2) Added a newline after messages like `Adding Storybook support to your "React" app`, since otherwise npm overwrote them once it started installs.  

The logs now look something like this:

```
▶ npx sb@next init

 sb init - the simplest way to add a Storybook to your project.

 • Detecting project type. ✓
    Detected vite project, setting builder to @storybook/builder-vite
 • Adding Storybook support to your "React" app
(#########⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠋ idealTree:@storybook/builder-vite: sill fetch manifest unfetch@^4.2.0
```

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

Without some vite examples for e2e tests, I'm not sure how to write automated tests for this.  I did test it out locally, though, and it worked great.

<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this. For additional guidance: https://storybook.js.org/docs/react/contribute/how-to-contribute

Maintainers: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

-->

IanVS committed:
Detect vite project, use vite builder automatically

IanVS committed:
Add newlines to avoid message being overwritten by npm

IanVS committed:
Remove community vite builder message

IanVS commented:
Hmmm, I can't tell why nx is failing here.  `yarn bootstrap --core` works fine on my own machine...

shilman reviewed (approved):
LGTM -- great stuff @IanVS ❤️

shilman committed:
trigger new CI build

IanVS committed:
Fix unit test

These additional methods are used in find-up and must be present in the mock `fs`
